test/model/m_competitions/test_predictors.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import tempfile
+from pathlib import Path
+
+# Third-party imports
+import pytest
+from flaky import flaky
+import pandas as pd
+import numpy as np
+from pydantic import PositiveInt
+
+# First-party imports
+from gluonts.dataset.artificial import constant_dataset
+from gluonts.evaluation.backtest import backtest_metrics
+from gluonts.evaluation import Evaluator
+from gluonts.model.predictor import Predictor
+from gluonts.model.m_competitions import (
+    Naive2Predictor,
+    SeasonalNaivePredictor,
+)
+from gluonts.dataset.common import Dataset
+from gluonts.support.pandas import forecast_start
+
+
+def generate_random_dataset(
+    num_ts: int, start_time: str, freq: str, min_length: int, max_length: int
+) -> Dataset:
+    start_timestamp = pd.Timestamp(start_time, freq=freq)
+    for _ in range(num_ts):
+        ts_length = np.random.randint(low=min_length, high=max_length)
+        target = np.random.uniform(size=(ts_length,))
+        data = {"target": target, "start": start_timestamp}
+        yield data
+
+
+PREDICTION_LENGTH = PositiveInt(30)
+SEASON_LENGTH = PositiveInt(210)
+START_TIME = "2018-01-03 14:37:12"  # That's a Wednesday
+MIN_LENGTH = 300
+MAX_LENGTH = 400
+NUM_TS = 10
+
+
+@pytest.mark.parametrize(
+    "RepresentablePredictor", [SeasonalNaivePredictor, Naive2Predictor]
+)
+@pytest.mark.parametrize(
+    "freq", ["1min", "15min", "30min", "1H", "2H", "12H", "7D", "1W", "1M"]
+)
+def test_seasonal_naive(RepresentablePredictor, freq: str):
```

Comment:
The argument name should not be camel-case, as that can be confused for a class name (and notice how GitHub indeed highlights it).

I suggest renaming `RepresentablePredictor` as `predictor_cls`